common/src/main/java/com/microsoft/identity/common/internal/ui/webview/challengehandlers/EoClientCertAuthChallengeHandler.java

```diff
@@ -0,0 +1,132 @@
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.internal.ui.webview.challengehandlers;
+
+import android.annotation.TargetApi;
+import android.app.Activity;
+import android.os.Build;
+import android.security.KeyChain;
+import android.security.KeyChainAliasCallback;
+import android.security.KeyChainException;
+import android.util.Log;
+import android.webkit.ClientCertRequest;
+
+import androidx.annotation.NonNull;
+
+import com.microsoft.identity.common.logging.Logger;
+import com.yubico.yubikit.android.YubiKitManager;
+import com.yubico.yubikit.android.transport.usb.UsbConfiguration;
+import com.yubico.yubikit.android.transport.usb.UsbYubiKeyDevice;
+import com.yubico.yubikit.core.util.Callback;
+
+import java.security.Principal;
+import java.security.PrivateKey;
+import java.security.cert.X509Certificate;
+
+public class EoClientCertAuthChallengeHandler implements IChallengeHandler<ClientCertRequest, Void> {
+    private static final String TAG = EoClientCertAuthChallengeHandler.class.getSimpleName();
+    private static final String ACCEPTABLE_ISSUER = "CN=MS-Organization-Access";
+    private Activity mActivity;
+    private final YubiKitManager mYubiKitManager;
+
+    public EoClientCertAuthChallengeHandler(@NonNull final Activity activity) {
+        mActivity = activity;
+        //Create and start YubiKitManager
+        mYubiKitManager = new YubiKitManager(mActivity.getApplicationContext());
+        mYubiKitManager.startUsbDiscovery(new UsbConfiguration(), new Callback<UsbYubiKeyDevice>() {
+            @Override
+            public void invoke(UsbYubiKeyDevice device) {
+                Log.i(TAG, "A YubiKey device was connected");
+                device.setOnClosed(new Runnable() {
+                    @Override
+                    public void run() {
+                        Log.i(TAG, "A YubiKey device was disconnected");
+                    }
+                });
+            }
+        });
+
+    }
+    //processChallenge is the same for now.
```

Comment:
If it is the same then we don't need to duplicate all this code. Let's create an `AbstractChallengeHandler` and move this code there so it can be shared between both EoClientCertChallengeHandler as well as the regular ClientCertChallengeHandler.

Reply:
Is it possible to not make ClientCertAuthChallengeHandler final, and then have EoClientCertAuthChallengeHandler extend it? Asking since ClientCertAuthChallengeHandler would basically be AbstractChallengeHandler, since EoClientCertAuthChallengeHandler will handle on-device CBA in the same way. 
My plan was to have ClientCertAuthChallengeHandler have a method that takes care of on-device CBA and call it in processChallenge. Then, EoClientCertAuthChallengeHandler can inherit this method and call it when it needs to (when a smartcard is not plugged in).

Reply:
I think it should be okay to remove `final` from that class.

Reply:
Latest commit implements the plan described above.